metpy/calc/basic.py

```diff
@@ -773,6 +773,163 @@ def smooth_n_point(scalar_grid, n=5, passes=1):
     return smooth_grid
 
 
+@exporter.export
+@preprocess_xarray
+def altimeter_to_station_pressure(altimeter_value, height):
+    r"""Convert the altimeter measurement to station pressure.
+
+    This function is useful for working with METARs since most will provide
+    altimeter values, but not sea-level pressure or station pressure.
+    The following definitions of altimeter setting and station pressure
+    are taken from the Federal Meteorology Handbook (2013) p. 11-1.
+    Altimeter setting is the pressure value to which an aircraft altimeter scale
+    is set so that it will indicate the altitude above mean sea-level of an aircraft
+    on the ground at the location for which the value is determined. It assumes a
+    standard atmosphere. Station pressure is the atmospheric pressure at the
+    designated station elevation. Finding the station pressure can be helpful for
+    calculating sea-level pressure or other parameters.
+
+    Parameters
+    ----------
+    altimeter_value : `pint.Quantity`
+            The altimeter setting value as defined by the METAR or other observation,
+            which can be measured in either inches of mercury (in. Hg) or millibars (mb)
+    height: `pint.Quantity`
+            Elevation of the station measuring pressure.
+
+    Returns
+    --------
+    station_pressure: `pint.Quantity`
+            The station pressure in hPa, which can be used to calculate sea-level pressure
+
+    See Also
+    ---------
+    altimeter_to_sea_level_pressure
+
+    Notes
+    -------
+    This function is implemented using the following equations from the
+    Smithsonian Handbook (1951) p. 269
+
+    Equation 1
+    .. math:: A_{mb} = (p_{mb} - 0.3)F
+
+    Equation 3
+    .. math:: F = \left [1 + \left(\frac{p_{0}^n a}{T_{0}} \right)
+    \frac{H_{b}}{p_{1}^n} \right ] ^ \frac{1}{n}
+
+    Where
+
+    p_{0} = standard sea-level pressure = 1013.25 mb
+
+    p_{1} = p_{mb} - 0.3 when p_{0} = 1013.25 mb
+
+    a = lapse rate in NACA standard atmosphere below the isothermal layer 0.0065
+       ^{\circ}C. m.^{-1})
+
+    T_{0} = standard sea-level temperature 288 K
+
+    H_{b} = station elevation in meters (elevation for which station pressure is given)
+
+    n = \frac{a R_{d}}{g} = 0.190284 where R_{d} is the gas constant for dry air
+
+    And solving for p_{mb} results in the equation below, which is used to
+    calculate station pressure (p_{mb})
+
+    .. math:: p_{mb} = \left [A_{mb} ^ n - \left (\frac{p_{0} a H_{b}}{T_0}
+    \right) \right] ^ \frac{1}{n} + 0.3
+
+    """
+    # Make sure the input values have the correct units
+    altim = altimeter_value.to('hPa')
```

Comment:
Why do we care what the units are? Shouldn't the unit support take care of this for us? (Should be simple enough to remove this line and see what the tests do.)